go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -227,21 +224,7 @@ func (agent *ActionAgent) InitMaster(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	// This section is similar to changeTypeLocked except that we don't
-	// invoke runHealthCheck. This is because the tabletserver cannot come
-	// up until the database is created, which happens after the replicas
-	// are pointed at this new master.
-	startTime := time.Now()
-	_, err = topotools.ChangeType(ctx, agent.TopoServer, agent.TabletAlias, topodatapb.TabletType_MASTER, logutil.TimeToProto(startTime))
-	if err != nil {
-		return "", err
-	}
-	// We only update agent's masterTermStartTime if we were able to update the topo.
-	// This ensures that in case of a failure, we are never in a situation where the
-	// tablet's timestamp is ahead of the topo's timestamp.
-	agent.setMasterTermStartTime(startTime)
-	// and refresh our state
-	if err := agent.refreshTablet(ctx, "InitMaster"); err != nil {
+	if err := agent.changeTypeLocked(ctx, topodatapb.TabletType_MASTER); err != nil {
```

Comment:
Doesn't this violate the caveat that the comment was documenting?

Reply:
This was complicated. The TL:DR is that this function should not `broadcastHealth`. Here's the long explanation:

I need to study more about how this can be simplified. What is happening is that the `refreshTablet` notices that the healthy state didn't change. And so, it does not invoke `broadcastHealth`.

The previous version of `changeTypeLocked` was invoking `runHealthCheck` which was unconditionally broadcasting. Changing this function to invoke `changeTypeLocked` ended up broadcasting to the vtgates that the tablet is unhealthy. The net effect of this was that the local example started failing because the tablets were non-serving for a few seconds longer after the scripts brought them up.

This was the reason why I could not change this function to call `changeTypeLocked` until I removed its invocation of `runHealthCheck`.

Reply:
@enisoc were you referring to the masterTermStartTime comment?

Reply:
This lgtm now that changeTypeLocked no longer calls runHealthCheckLocked.